### PR TITLE
[PAN-2852] privacy group id consistent

### PR DESCRIPTION
**Pull request by iikirilov**
# PR description
Handle base64 <-> BytesValue conversion uniformly throughout Pantheon

As a side effect also fixes the private transaction gas required to 23992 as specified in [PAN-2795]

## Fixed Issue(s)
[PAN-2852]

[PAN-2795]: https://pegasys1.atlassian.net/browse/PAN-2795
[PAN-2852]: https://pegasys1.atlassian.net/browse/PAN-2852

### CODE REVIEW

File: acceptance-tests/src/test/java/tech/pegasys/pantheon/tests/web3j/privacy/EnclaveErrorAcceptanceTest.java [hunk 1]
```diff
@@ -32,7 +32,7 @@
 import org.junit.Before;
 import org.junit.Test;
 
-public class PrivateTxEnclaveErrorAcceptanceTest extends PrivacyAcceptanceTestBase {
+public class EnclaveErrorAcceptanceTest extends PrivacyAcceptanceTestBase {
```
Comment (shemnon) [inline acceptance-tests/src/test/java/tech/pegasys/pantheon/tests/web3j/privacy/EnclaveErrorAcceptanceTest.java]: Did this file get renamed?  It's unclear from the Github commit.

File: ethereum/core/src/main/java/tech/pegasys/pantheon/ethereum/mainnet/precompiles/privacy/PrivacyPrecompiledContract.java [hunk 2]
```diff
@@ -90,7 +88,7 @@ public void setPrivateTransactionProcessor(
 
   @Override
   public Gas gasRequirement(final BytesValue input) {
-    return Gas.of(40_000L); // Not sure
+    return Gas.of(0);
```
Comment (shemnon) [inline ethereum/core/src/main/java/tech/pegasys/pantheon/ethereum/mainnet/precompiles/privacy/PrivacyPrecompiledContract.java]: Is this the EEA specified gas?

File: ethereum/jsonrpc/src/main/java/tech/pegasys/pantheon/ethereum/jsonrpc/internal/methods/privacy/EeaDeletePrivacyGroup.java [hunk 3]
```diff
@@ -46,8 +46,8 @@ public String getName() {
   public JsonRpcResponse response(final JsonRpcRequest request) {
     LOG.trace("Executing {}", RpcMethod.EEA_DELETE_PRIVACY_GROUP.getMethodName());
 
-    final String privacyGroupId = parameters.required(request.getParams(), 1, String.class);
```
Comment (shemnon) [inline ethereum/jsonrpc/src/main/java/tech/pegasys/pantheon/ethereum/jsonrpc/internal/methods/privacy/EeaDeletePrivacyGroup.java]: This seems like a non-change.  If there are any other alterations can this be reverted?

File: ethereum/jsonrpc/src/main/java/tech/pegasys/pantheon/ethereum/jsonrpc/internal/methods/privacy/EeaGetTransactionCount.java [hunk 4]
```diff
@@ -51,6 +51,7 @@ public JsonRpcResponse response(final JsonRpcRequest request) {
     final String privacyGroupId = parameters.required(request.getParams(), 1, String.class);
 
     final long nonce = privateTransactionHandler.getSenderNonce(address, privacyGroupId);
+
```
Comment (shemnon) [inline ethereum/jsonrpc/src/main/java/tech/pegasys/pantheon/ethereum/jsonrpc/internal/methods/privacy/EeaGetTransactionCount.java]: Again, another non-change.

File: acceptance-tests/src/test/java/tech/pegasys/pantheon/tests/web3j/privacy/EnclaveErrorAcceptanceTest.java (hunk 1 above)
Comment (iikirilov) [reply acceptance-tests/src/test/java/tech/pegasys/pantheon/tests/web3j/privacy/EnclaveErrorAcceptanceTest.java]: Yes, the file was renamed. It should be `PrivateTransactionEnclaveErrorAcceptanceTest` but that is a mouthful so I have gone with `EnclaveErrorAcceptanceTest` which describes the test well

File: ethereum/core/src/main/java/tech/pegasys/pantheon/ethereum/mainnet/precompiles/privacy/PrivacyPrecompiledContract.java (hunk 2 above)
Comment (iikirilov) [reply ethereum/core/src/main/java/tech/pegasys/pantheon/ethereum/mainnet/precompiles/privacy/PrivacyPrecompiledContract.java]: There is no requirement in the EEA spec. The reasoning here is that implementing EEA compliant private transaction support without a Precompile contract would cost 21000 gas like a normal transaction. This precompile should not consume gas.

**Pull request merged**